packages/cli/src/ui/utils/kittyProtocolDetector.ts

```diff
@@ -107,13 +108,17 @@ export function isKittyProtocolEnabled(): boolean {
 }
 
 function disableAllProtocols() {
-  if (kittyEnabled) {
-    process.stdout.write('\x1b[<u');
-    kittyEnabled = false;
-  }
-  if (sgrMouseEnabled) {
-    process.stdout.write('\x1b[?1006l'); // Disable SGR Mouse
-    sgrMouseEnabled = false;
+  try {
+    if (kittyEnabled) {
```

Comment:
add a comment why we use the nonstandard writeSync which would generally be against project guidelines if it wasn't for the reasons mentioned in the pr description. You could even just mention the bug this is believed to fix.